### PR TITLE
msgpack-cxx 4.0.1 (new formula)

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

---

it failed to pass the audit check

```
msgpack-cpp:
  * HEAD: Use `branch: "master"` to specify the default branch
```

We might want to have a allowlist for branch audit check.

### CODE REVIEW

File: Formula/msgpack-cpp.rb
Comment (carlocab) [inline Formula/msgpack-cpp.rb:9]: I think this is needed at runtime.

File: Formula/msgpack.rb
Comment (carlocab) [inline Formula/msgpack.rb:20]:
```suggestion
```
This is not true. `msgpack` does not ship any binaries.
```
❯ brew ls -l msgpack
/usr/local/Cellar/msgpack/3.3.0/include/msgpack/ (737 files)
/usr/local/Cellar/msgpack/3.3.0/include/ (2 files)
/usr/local/Cellar/msgpack/3.3.0/lib/libmsgpackc.2.0.0.dylib
/usr/local/Cellar/msgpack/3.3.0/lib/cmake/ (4 files)
/usr/local/Cellar/msgpack/3.3.0/lib/pkgconfig/msgpack.pc
/usr/local/Cellar/msgpack/3.3.0/lib/ (3 other files)
```

File: Formula/msgpack-cpp.rb
Comment (carlocab) [inline Formula/msgpack-cpp.rb:1]: Should we call this `msgpack-cxx`?

File: Formula/msgpack-cpp.rb
Comment (carlocab) [inline Formula/msgpack-cpp.rb:12]: I think the conflict comes from both installing `include/msgpack`. Perhaps we can rename the one here to `include/msgpack-cxx`?
```ruby
include.install include/"msgpack" => "msgpack-cxx"
```
That we we won't need the conflict.

File: Formula/msgpack-cpp.rb
Comment (chenrui333) [reply Formula/msgpack-cpp.rb:1]: I saw most of the formulae used `cpp` part of the names, also the head branch is also with `cpp_master`

File: Formula/msgpack-cpp.rb
Comment (chenrui333) [reply Formula/msgpack-cpp.rb:12]: ok, let's do that.

**Comment by chenrui333** [comment]
Considering most of the formulae are with name of `cpp` as well as the head branch is with `cpp_master`, we should use `msgpack-cpp`, but the artifact is with `msgpack-cxx` though.

File: Formula/msgpack-cpp.rb
Comment (chenrui333) [reply Formula/msgpack-cpp.rb:9]: that sounds about right!

File: Formula/msgpack-cpp.rb
Comment (chenrui333) [reply Formula/msgpack-cpp.rb]: The out-of-the-box example does not work with this change.

File: Formula/msgpack.rb
Comment (carlocab) [inline Formula/msgpack.rb:7]: Let's do this in a separate PR so we don't have to rebuild bottles for this and throw away the `high_sierra` bottle.

File: Formula/msgpack-cxx.rb
Comment (carlocab) [inline Formula/msgpack-cxx.rb:25]:
```suggestion
      #include <msgpack-cxx.hpp>
```

Comment (chenrui333) [reply Formula/msgpack-cxx.rb:25]: With that change, this is the error in my local
```
In file included from test.cpp:1:
In file included from /usr/local/Cellar/msgpack-cxx/4.0.1/include/msgpack-cxx.hpp:10:
/usr/local/Cellar/msgpack-cxx/4.0.1/include/msgpack-cxx/object.hpp:13:10: fatal error: 'msgpack/object_decl.hpp' file not found
#include "msgpack/object_decl.hpp"
         ^~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
Error: msgpack-cxx: failed
```

File: Formula/msgpack.rb
Comment (chenrui333) [reply Formula/msgpack.rb:7]: Sounds good.

Comment (chenrui333) [reply Formula/msgpack.rb]: Moved into #84264

File: Formula/msgpack-cxx.rb
Comment (carlocab) [inline Formula/msgpack-cxx.rb:19]: Oops. Looks like renaming didn't work. Let's drop the renaming, and add the conflict in a follow-up PR so we can keep the `msgpack` bottle for High Sierra.

Comment (carlocab) [reply Formula/msgpack-cxx.rb:19]: Yup, don't add `conflicts_with` for now. Let's add it right after this is merged.

Comment (chenrui333) [reply Formula/msgpack-cxx.rb:19]: I see your poiint.

**Review by carlocab — APPROVED** [review]
Failures on ARM and Linux are audit failures for the `head` branch, but those are wrong. Assuming no build failures on Intel macOS, this should be good to merge.

**Pull request closed without merging**
